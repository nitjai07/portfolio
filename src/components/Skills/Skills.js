import React, { Component } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import { withRouter, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Heading from '../Common/Heading/Heading';
// import meImgLow from '../../assets/images/me_1_low.jpg';

class Skills extends Component {
  componentDidMount() {
    setTimeout(() => {
      window.addEventListener('mousewheel', this.handleScroll);
    }, 500);
  }

  componentWillUnmount() {
    window.removeEventListener('mousewheel', this.handleScroll);
  }

  handleScroll = event => {
    if (event.deltaY > 30 && window.innerHeight + window.scrollY >= document.body.offsetHeight + 99) {
      this.props.history.push('./projects');
    } else if (window.scrollY === 0 && event.deltaY < -30) {
      this.props.history.push('./about');
    }
  };

  render() {
    return (
      <ReactCSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={400}
        transitionEnterTimeout={400}
        transitionLeaveTimeout={100}
        transitionName="loadComponent"
      >
        <div className="skillswrapper">
          <div className="skillswrapper__leftsec">
            <Heading headingText="Skills & Experience" />
            <p className="skillswrapper__content">
              Main area of my expertise is front end development and everything related with this side of web. Mainly i
              work on HTML, CSS, Javascript, Angular JS, SASS. I like working on projects from scratch and have good end
              to end development understanding.
              <br /> <br /> I have also some expertise with Node.js, Mongo DB and React.
              <br /> <br /> Would like to know more?
              <br /> Please check my&nbsp;
              <a href="https://www.linkedin.com/in/nitish-jain-59b59a35/" className="linkdefault" target="blank">
                LinkedIn
              </a>
              &nbsp;profile or just&nbsp;
              <Link to="/contact" className="linkdefault">
                Contact
              </Link>
              &nbsp;me.
            </p>
          </div>
          <div className="stage">
            <div className="cubespinner">
              <div className="face1">
                <FontAwesomeIcon icon={['fab', 'react']} />
              </div>
              <div className="face2">
                <FontAwesomeIcon icon={['fab', 'angular']} />
              </div>
              <div className="face3">
                <FontAwesomeIcon icon={['fab', 'sass']} />
              </div>
              <div className="face4">
                <FontAwesomeIcon icon={['fab', 'html5']} />
              </div>
              <div className="face5">
                <FontAwesomeIcon icon={['fab', 'node']} />
              </div>
              <div className="face6">
                <FontAwesomeIcon icon={['fab', 'accessible-icon']} />
              </div>
            </div>
          </div>
        </div>
      </ReactCSSTransitionGroup>
    );
  }
}

Skills.propTypes = {
  history: PropTypes.object
};

Skills.defaultProps = {
  history: ''
};

export default withRouter(Skills);
