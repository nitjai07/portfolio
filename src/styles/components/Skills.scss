.skillswrapper {
  &__leftsec {
    display: inline-block;
    width: 45%;
    color: $secondaryColor2;
  }
}

@keyframes spincube {
  from,to {
    transform: rotateX(0deg) rotateY(0deg) rotateZ(0deg);
  }
  16% {
    transform: rotateY(-90deg);
  }
  33% {
    transform: rotateY(-90deg) rotateZ(90deg);
  }
  50% {
    transform: rotateY(-180deg) rotateZ(90deg);
  }
  66% {
    transform: rotateY(-270deg) rotateX(90deg);
  }
  83% {
    transform: rotateX(90deg);
  }
}

.stage {
  width: 50%;
  display: inline-block;
  position: fixed;
  top: 40%;
  left: 70%;
}

.cubespinner {
  animation-name: spincube;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;
  animation-duration: 12s;
  transform-style: preserve-3d;
  transform-origin: 60px 60px 0;
}

.cubespinner div {
  position: absolute;
  width: 150px;
  height: 150px;
  border: 1px solid $secondaryColor2;
  background: rgba($primaryColor1, 0.8);
  box-shadow: inset -10px 10px 40px $secondaryColor1;
  line-height: 150px;
  text-align: center;
  font-size: 100px;
  backface-visibility: hidden;
}

.cubespinner .face1 {
  transform: translateZ(75px); 
}
.cubespinner .face2 {
  transform: rotateY(90deg) translateZ(75px);
}
.cubespinner .face3 {
  transform: rotateY(90deg) rotateX(90deg) translateZ(75px);
}
.cubespinner .face4 {
  transform: rotateY(180deg) rotateZ(90deg) translateZ(75px);
}
.cubespinner .face5 {
  transform: rotateY(-90deg) rotateZ(90deg) translateZ(75px);
}
.cubespinner .face6 {
  transform: rotateX(-90deg) translateZ(75px);
}