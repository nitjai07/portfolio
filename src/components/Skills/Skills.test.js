import React from 'react';
import { shallow } from 'enzyme';
import Skills from './Skills';

test('Skills Component renders correctly', () => {
  const component = shallow(<Skills />);
  expect(component).toMatchSnapshot();
});
