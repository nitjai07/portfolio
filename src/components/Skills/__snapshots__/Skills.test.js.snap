// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Route>
    [Function]
</Route>,
  "nodes": Array [
    <Route>
      [Function]
</Route>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <withRouter(Skills) />,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <Route>
      [Function]
</Route>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <withRouter(Skills) />,
}
`;
